### PR TITLE
Run integration tests in browser

### PR DESCRIPTION
**Pull request by fealebenpae**

### CODE REVIEW

File: packages/realm/bindgen/src/realm_js_browser_helpers.h [hunk 1]
```diff
@@ -16,18 +16,54 @@ REALM_NOINLINE inline emscripten::val toEmscriptenErrorCode(const std::error_cod
...
+    const view = Emval.toValue(view_handle);
+    view.set(new Uint8Array(buffer));
+});
+// Allocate a new C++ buffer big enough to fit the JS buffer
+// Create a JS memory view around the C++ buffer
+// Call TypedArray.prototype.set to efficiently copy the JS buffer into the C++ buffer via the view
+REALM_NOINLINE inline std::string toBinaryData(const emscripten::val array_buffer) noexcept
+{
+    REALM_ASSERT(array_buffer.instanceof(emscripten::val::global("ArrayBuffer")));
+    std::string buf;
```
Comment (nhachicha) [inline packages/realm/bindgen/src/realm_js_browser_helpers.h:46]: This acts as the C++ buffer, shouldn't this be heap allocated?

File: packages/realm/package.json [hunk 2]
```diff
@@ -114,7 +115,11 @@
       "command": "cmake-js configure --debug --directory bindgen"
     },
     "bindgen:configure:browser": {
-      "command": "emcmake cmake -DCMAKE_BUILD_TYPE=Debug -S bindgen -B bindgen/build.browser -G Ninja"
+      "command": "emcmake cmake -DCMAKE_BUILD_TYPE=Debug -S bindgen -B bindgen/build.browser -G Ninja",
+      "files": [
```
Comment (nhachicha) [inline packages/realm/package.json:119]: Did this fixed the caching issues with `wireit`?

**Review by nhachicha — APPROVED** [review]

File: packages/realm/bindgen/browser.cmake [hunk 3]
```diff
@@ -1,4 +1,4 @@
 add_executable(realm-js-wasm)
-target_link_options(realm-js-wasm PRIVATE -d -sALLOW_MEMORY_GROWTH=1 -sLLD_REPORT_UNDEFINED -sFETCH=1 -lembind -fwasm-exceptions -sEXPORT_ES6=1 -sWASM_BIGINT=1 -sENVIRONMENT=web --pre-js=../web_polyfill.js)
+target_link_options(realm-js-wasm PRIVATE -d -sALLOW_MEMORY_GROWTH=1 -sLLD_REPORT_UNDEFINED -sFETCH=1 -lembind -fwasm-exceptions -sEXPORT_ES6=1 -sWASM_BIGINT=1 -sENVIRONMENT=web -sSTACK_SIZE=131072 --pre-js=../web_polyfill.js)
```
[the remark this answers is not in the document]
Comment (fealebenpae) [reply packages/realm/bindgen/browser.cmake:2]: I just got stack overflow errors that advised me to set `STACK_SIZE` and I just doubled the default 64kb to 128kb.

File: packages/realm/bindgen/src/realm_js_browser_helpers.h (hunk 1 above)
Comment (fealebenpae) [reply packages/realm/bindgen/src/realm_js_browser_helpers.h:46]: It eventually gets moved to the string buffers deque in `RealmAddon` - I need to double-check there aren't any extra copies along the way, but the eventual `BinaryData` we construct points to the buffer in the deque, not the stack variable here.

File: packages/realm/package.json (hunk 2 above)
Comment (fealebenpae) [reply packages/realm/package.json:119]: Yeah, thankfully I haven't had to delete it manually while iterating on this.

**Pull request merged**